api/node/typescript/index.ts

```diff
@@ -614,9 +614,14 @@ function loadSlint(loadData: LoadData): Object {
  *          For further information on the available properties, refer to [Instantiating A Component](../index.html#md:instantiating-a-component).
  * @throws {@link CompileError} if errors occur during compilation.
  */
-export function loadFile(filePath: string, options?: LoadFileOptions): Object {
+export function loadFile(
+    filePath: string | URL,
+    options?: LoadFileOptions,
+): Object {
+    const pathname =
+        typeof filePath === "string" ? filePath : filePath.pathname;
```

Comment:
I suggest to write this the other way around. If you know for sure that it's a `URL`, then extract the `pathname` property, otherwise assume that it's a "string like". This way it works with types that can be implicitly converted as well.